### ISSUE TITLE
Issue when RSS feed has only 1 item

### ISSUE THREAD

**Issue by cfusiondev** [amzn/fire-app-builder#48]
When the RSS feed has only one item I get the following error: Recipe chain failed:
    java.lang.ClassCastException: java.lang.String cannot be cast to java.util.ArrayList

Is this a bug or am I missing something in the setup?

**Comment by cfusiondev** [comment]
This is the feed xml:

<rss xmlns:media="http://search.yahoo.com/mrss/" xmlns:content="http://purl.org/rss/1.0/modules/content/" version="2.0">
--
  | <channel>
  | <link>
  | https://www.riverbender.com
  | </link>
  | <language>en-us</language>
  | <pubDate>Mon, 08 Apr 2019 15:25:56 EST</pubDate>
  | <title>RiverBender Fire Stick App</title>
  | <description>Video for Fire Stick</description>
  | <lastBuildDate>Mon, 08 Apr 2019 15:25:56 EST</lastBuildDate>
  |  
  |  
  | <item>
  | <title> Judges 21:1-25 - The Bridge Service - April 7th, 2019</title>
  | <description> Service at the Bridge on April 7th, 2019. Sermon on Judges 21:1-25. Services are held every Sunday at 10 AM at 504 E 12th Street in Alton IL. www.thebridgealton.com</description>
  | <category>Community</category>
  | <media:content url="https://s3.amazonaws.com/riverbender/video/3543.mp4" type="video/mp4" medium="video" duration="5814">
  | <media:description type="plain"> Service at the Bridge on April 7th, 2019. Sermon on Judges 21:1-25. Services are held every Sunday at 10 AM at 504 E 12th Street in Alton IL. www.thebridgealton.com</media:description>
  | <media:thumbnail url="https://s3.amazonaws.com/riverbender/videoThumbs/3543.jpg" />
  | </media:content>
  | <guid isPermaLink="false">Video_3543</guid>
  | <pubDate>Mon, 08 Apr 2019 09:19:13 EST</pubDate>
  | </item>
  |  
  | </channel>
  | </rss>


This is the recipe:

{
  "cooker": "DynamicParser",
  "format": "xml",
  "model": "com.amazon.android.model.content.Content",
  "translator": "ContentTranslator",
  "modelType": "array",
  "query": "//item[./category='$$par0$$']",
  "matchList": [
    "title/#text@mTitle",
    "guid/#text@mId",
    "description/#text@mDescription",
    "media:content/#attributes/url@mUrl",
    "media:content/media:thumbnail/#attributes/url@mCardImageUrl",
    "media:content/media:thumbnail/#attributes/url@mBackgroundImageUrl"
  ]
}

If I allow 2 items, it works, however, if the 3rd item has a different category it will fail.

**Comment by levonlevonian** [comment]
Thanks, looking into it...

**Comment by TheFloridaChannel** [comment]
This is exactly what we've experienced and reported back in November 2018. https://github.com/amzn/fire-app-builder/issues/28 will there be any updates on this soon? It's a pretty huge issue and we're not able to deploy our app because of it. We have also logged a few more potential bugs via the Amazon developer portal but we do not get responses. Case 5879379321 for reference.

**Comment by levonlevonian** [comment]
Hi there,

Yes, we remember the issue with a single item that you raised earlier, and please rest assured that the Fire App Builder team is aware and is working on it too, among other things. I don't have an ETA, but it's planned to be released in one of the upcoming versions. Thank you!

**Comment by cfusiondev** [comment]
So... we just wait for the next release to come out when?

**Comment by levonlevonian** [comment]
Correct. Regrettably, I don't have any estimates for when, but please keep an eye on any updates. Thanks!

**Comment by TheFloridaChannel** [comment]
well at the very least, we appreciate your response because communication around this issue, and the others we've found that persist, have been.. Well, let's say lacking. Thank you.

**Comment by cfusiondev** [comment]
Thank you, I'll just come up with a workaround on our end.
